### PR TITLE
Fix possible notification loop (EXPOSUREAPP-4036)

### PR DESCRIPTION
**Pull request by d4rken**
This fixes the issue that if no new risk level calculation happened, and the last one triggered a notification, killing and restarting the app would also trigger a notification.

To test this you'd have to get to a state where the last risk level calculations were LOW_RISK -> HIGH_RISK, and then just kill the app process and open the app again.

Also: While debugging this I noticed a few cryptic log messages which needed explicit tags.

### CODE REVIEW

**Review by chris-cwa — APPROVED** [review]

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/risk/RiskLevelChangeDetector.kt [hunk 1]
```diff
@@ -47,6 +48,13 @@ class RiskLevelChangeDetector @Inject constructor(
         val oldResult = changedLevels.first()
         val newResult = changedLevels.last()
 
+        val lastCheckedResult = riskLevelSettings.lastChangeCheckedRiskLevelTimestamp
+        if (lastCheckedResult == newResult.calculatedAt) {
+            Timber.d("We already checked this risk level change, skipping further checks.")
+            return
+        }
+        riskLevelSettings.lastChangeCheckedRiskLevelTimestamp = newResult.calculatedAt
+
```
Comment (ritsam) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/risk/RiskLevelChangeDetector.kt:57]: is there any advantage to create the variable _lastCheckedResult_ ? can we not directly compare _riskLevelSettings.lastChangeCheckedRiskLevelTimestamp_ in the condition?

The comment is more for an information seeking purpose rather that a suggestion.

```suggestion
        if (iskLevelSettings.lastChangeCheckedRiskLevelTimestamp == newResult.calculatedAt) {
            Timber.d("We already checked this risk level change, skipping further checks.")
            return
        }
        riskLevelSettings.lastChangeCheckedRiskLevelTimestamp = newResult.calculatedAt

```

Comment (d4rken) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/risk/RiskLevelChangeDetector.kt:58]: Readability, inlining the variable exceeds our maximum line length.

**Review by BMItr — APPROVED** [review]
code lgtm, nothing to complain.
passed local testing
image

**Review by ralfgehrer — COMMENTED** [review]
Code looks fine. Testing now. Should also fix another issue (EXPOSUREAPP-4050)

**Review by ralfgehrer — APPROVED** [review]
works like a charm

**Pull request merged**